diff --git a/source/common/IntraPred.cpp b/source/common/intrapred.cpp
similarity index 100%
rename from source/common/IntraPred.cpp
rename to source/common/intrapred.cpp
diff --git a/source/common/x86/x86inc.asm b/source/common/x86/x86inc.asm
index dde6b7e0b..d13cdcf44 100644
--- a/source/common/x86/x86inc.asm
+++ b/source/common/x86/x86inc.asm
@@ -64,6 +64,15 @@
     %else
         %define UNIX64 1
     %endif
+%endif
+ 
+%define FORMAT_ELF 0
+%ifidn __OUTPUT_FORMAT__,elf
+    %define FORMAT_ELF 1
+%elifidn __OUTPUT_FORMAT__,elf32
+    %define FORMAT_ELF 1
+%elifidn __OUTPUT_FORMAT__,elf64
+    %define FORMAT_ELF 1
 %endif
 
 %define FORMAT_ELF 0
